lib/paperclip/storage/database.rb

```diff
@@ -108,10 +108,11 @@ def setup_paperclip_file_model
 
       def setup_attachment_class
         instance.class.ancestors.each do |ancestor|
+          next unless ancestor.respond_to?(:attachment_definitions)
+
           # Pick the top-most definition like
           # Paperclip::AttachmentRegistry#definitions_for
-          names_for_ancestor = ancestor.attachment_definitions.keys rescue []
-          if names_for_ancestor.member?(name)
+          if ancestor.attachment_definitions.member?(name)
             @attachment_class = ancestor
```

Comment:
Can't we break after first definition found? Why comment says that we are looking for top-most definition? How about class `A` defining attachment and class `B < A` overriding attachment definition? In this case I think we should get definitions from class `B` but this code will find definition from `A`.